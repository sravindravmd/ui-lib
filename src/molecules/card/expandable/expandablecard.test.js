import React from 'react'
import { render, cleanup, fireEvent } from '@testing-library/react'
import ExpandableCard from './index'
import CardItem from '../carditem'

afterEach(cleanup)

describe('<ExpandableCard/>', () => {
  it('renders ExpandableCard correctly', () => {
    const { container } = render(
      <ExpandableCard heading='Card'>
        <CardItem variant='detailItem' title='name3' property='property3' />
      </ExpandableCard>
    )
    expect(container).toMatchSnapshot()
  })

  it('expand card correctly', () => {
    const { getByTestId, getByText } = render(
      <ExpandableCard heading='Card'>
        <CardItem variant='detailItem' title='name3' property='property3' />
      </ExpandableCard>
    )
    const downArrow = getByTestId('downarrow')
    fireEvent.click(downArrow)
    expect(getByText('name3')).toBeInTheDocument()
  })
})
