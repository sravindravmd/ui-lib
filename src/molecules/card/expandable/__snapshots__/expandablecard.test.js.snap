// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExpandableCard/> renders ExpandableCard correctly 1`] = `
<div>
  <div
    class="sc-bdfBwQ fWZGSR"
  >
    <div
      class="sc-dlfnbm jYeWQn"
    >
      <div
        class="sc-hKgILt eLIWMT"
      >
        Card
      </div>
      <div
        class="sc-gsTCUz dAxUvf"
        data-testid="downarrow"
      >
        <svg
          fill="none"
          height="7"
          viewBox="0 0 12 7"
          width="12"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M0.519346 0.527549C0.807911 0.236494 1.27804 0.235307 1.56808 0.5249L6.47801 5.42743C6.7666 5.71557 6.76778 6.18285 6.48065 6.47245C6.19209 6.76351 5.72196 6.76469 5.43192 6.4751L0.521985 1.57257C0.233403 1.28443 0.232222 0.81715 0.519346 0.527549Z"
            fill="#121213"
            fill-rule="evenodd"
          />
          <path
            clip-rule="evenodd"
            d="M11.4807 0.527549C11.1921 0.236494 10.722 0.235307 10.4319 0.5249L5.52199 5.42743C5.2334 5.71557 5.23222 6.18285 5.51935 6.47245C5.80791 6.76351 6.27804 6.76469 6.56808 6.4751L11.478 1.57257C11.7666 1.28443 11.7678 0.81715 11.4807 0.527549Z"
            fill="#121213"
            fill-rule="evenodd"
          />
        </svg>
      </div>
    </div>
    <div
      class="sc-eCssSg boTWkf"
    >
      <div
        class="sc-fubCfw icXNwV"
      >
        <div
          class="sc-gKsewC gnEjpX"
        >
          name3
        </div>
        <div
          class="sc-iBPRYJ bthhOv"
        >
          property3
          <span
            class="sc-jSgupP kbRBxV"
          >
            <svg
              fill="none"
              height="12"
              viewBox="0 0 12 12"
              width="12"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M10.1207 0.5C10.322 0.5 10.5019 0.534424 10.6651 0.599143C10.8318 0.665267 10.9775 0.758527 11.1055 0.880373C11.2306 0.99953 11.3254 1.13664 11.3924 1.29517C11.4584 1.4513 11.4962 1.62903 11.5 1.83399C11.4997 1.9981 11.482 2.13089 11.4534 2.23703C11.4211 2.35663 11.3767 2.46508 11.3212 2.5639C11.263 2.66746 11.1903 2.77097 11.1014 2.87419C11.0957 2.87357 11.0864 2.87057 11.0774 2.86154L9.11944 0.906298C9.11633 0.903193 9.11484 0.900551 9.11397 0.898418C9.11298 0.895975 9.11259 0.893595 9.1126 0.891709C9.11261 0.891298 9.11262 0.890956 9.11265 0.890679C9.20423 0.805589 9.30027 0.735453 9.40085 0.678821C9.50346 0.621047 9.61006 0.576522 9.72129 0.54471C9.81334 0.518385 9.94349 0.5 10.1207 0.5ZM7.08433 2.92294C7.27931 2.72559 7.59767 2.72471 7.79374 2.92097L9.07302 4.2015C9.26819 4.39686 9.26803 4.71345 9.07267 4.90861L3.46161 10.514L1.61597 11.0295C1.23639 11.1355 0.889078 10.7812 1.00268 10.4038L1.5765 8.49766L7.08433 2.92294Z"
                fill="#121213"
                stroke="#121213"
              />
            </svg>
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
